components/modal/src/Modal.js

```diff
@@ -0,0 +1,114 @@
+/** @jsx jsx */
+
+import React, { Children, cloneElement, useState, useEffect } from 'react';
+import { jsx, useTheme } from '@westpac/core';
+import ReactDOM from 'react-dom';
+import PropTypes from 'prop-types';
+import shortid from 'shortid';
+import FocusLock from 'react-focus-lock';
+import { CSSTransition } from 'react-transition-group';
+import { ModalBackdrop, StyledModal } from './styled';
+import { ModalHeader } from './ModalHeader';
+import { ModalBody } from './ModalBody';
+
+// ==============================
+// Component
+// ==============================
+export const Modal = ({ isOpen, onClose, size, children, ...props }) => {
+	const [open, setOpen] = useState(isOpen);
+
+	useEffect(
+		() => {
+			setOpen(isOpen);
+		},
+		[isOpen]
+	);
+
+	const handleClose = () => {
+		if (onClose) {
+			onClose();
+		} else {
+			setOpen(false);
+		}
+	};
+
+	const modalId = shortid.generate();
+	const titleId = `modal-header-title-${modalId}`;
+	const bodyId = `modal-body-${modalId};`;
+
+	const childrenWithProps = Children.map(children, child => {
+		if (child && child.type) {
+			if (child.type === ModalHeader) {
+				return cloneElement(child, {
+					id: titleId,
+					onClose: handleClose,
+				});
+			}
+
+			if (child.type === ModalBody) {
+				return cloneElement(child, {
+					id: bodyId,
+				});
+			}
+		}
+		return child;
+	});
```

Comment:
Can/should this be refactored as one return and cloneElement call? 🤔

This is probs easier to follow?

Reply:
Refactoring it to one return in this case would involve more logic and nested ternaries making it harder to read.

Reply:
Yeah I had a go. Came out longer 😢